### ISSUE TITLE
Cleanup / Shutdown Plugin

### ISSUE THREAD

**Issue by msacket** [scullyio/scully#304]
### Description

It would be nice if we could write a plugin (or hook an existing plugin) into the cleanup/shutdown phase of Scully.  I'm working on a plugin that generates a sitemap.xml that could benefit from a cleanup hook.  Currently, the plugin re-writes the sitemap.xml as each route is rendered; however, I'd like to output the compiled map after all the route rendering is complete.

### Describe the solution you'd like

Add a `cleanup` or `onComplete` function to the `ScullyConfig`.  These functions gets called after all rendering is complete.  A plugin could export a cleanup/shutdown function like so:
```typescript
const { Sitemap , GenerateSitemap } = require('./dist/scully-plugin-sitemap');

exports.config = {
  projectRoot: "./src",
  projectName: "scully-plugins",
  defaultPostRenderers: [ Sitemap ],
  sitemapOptions: {},
  onComplete: [GenerateSitemap]
  outDir: './dist/static',
  routes: {
  }
};
```

### Describe alternatives you've considered

1) Currently, I re-generate the output as each route is rendered.  This works okay for a small static site; however, I suspect performance could be an issue on larger sites.

2) Add a shutdown parameter to the `registerPlugin` function.  (I'm not sure that's a great idea as it would be a breaking change and require updating existing plugins.)

3) Add a new `cleanup` plugin type.  It could utilize the same plugin management functionality and be registered the same as the other plugins.

4) Other options include traversing the generated hierarchy with another script via another build step.

**Comment by msacket** [comment]
If you think this is a good idea, I do have the code written for the `onComplete` handler.  I can submit a pull request if you want.

**Comment by SanderElias** [comment]
@msacket I do like the idea, but I would prefer to talk this through before venturing in.

**Comment by SanderElias** [comment]
This is solved by pr #390
